.github/workflows/e2e-test.yml

```
@@ -98,6 +98,19 @@ jobs:
         id: cypress-prep
         uses: ./.github/actions/prepare-cypress
 
+      - name: Download failed test artifacts
+        uses: actions/download-artifact@v4
+        continue-on-error: true
+        with:
+          pattern: failed-tests-${{ inputs.name }}-${{ inputs.edition }}
+          path: failed-tests
+
+      - name: Process failed tests
+        id: process-failed-tests
+        env:
+          TEST_GROUP: ${{ inputs.name }}-${{ inputs.edition }}
+        run: node .github/scripts/process-failed-tests.js
```

if we restructure the failed tests file, this can just become
```suggestion
        run: echo "specs=$(cat failed-tests-${{ inputs.name }}-${{ inputs.edition }}/failed-tests.txt)" >> $GITHUB_ENV
```